### ISSUE TITLE
Allow setup of S2I build/deployment using arbitrary image/code repository from add to project.

### ISSUE THREAD

**Issue by GrahamDumpleton** [openshift/origin-web-console#1021]
The add to project page of web console allows selecting from catalog (Browse Catalog), deployment from pasted in YAML/JSON (Import YAML/JSON), or deployment of image (Deploy Image).

The web console doesn't currently allow you to deploy source code where you nominate an arbitrary image to use as the source-to-image builder. At present the only similar option is to choose a built in S2I builder image from the catalog and then provide it with the code repository URL.

There should be a 'Deploy Source' option which mirrors the ability to select an image as in 'Deploy Image', but then in addition to that, requires you to provide a code repository URL to apply the selected image to, as a S2I builder.

If it mirrors the 'Deploy Image' page for image selection, this would allow you to select either a local image stream, or a remote image stored on a registry. In both cases the web console could validate that the image selected is in fact a S2I builder by looking at the labels on the image pulled down with the image description from the registry. If it doesn't validate as a S2I builder, show a warning, otherwise allow entry of the code repository to apply it to.

In short, the 'Deploy Source' page would mirror ``oc new-app source-image~repository-url`` functionality.

**Comment by jwforres** [comment]
This is considered part of this trello card https://trello.com/c/fonu3qc5

**Comment by GrahamDumpleton** [comment]
What I am describing is not mentioned at all in that Trello card as far as I can see. That Trello card talks about deploy buttons for external sites, or deploying from templates. It does not mention a 'Deploy Source' concept where from the UI you explicitly pick the image and then the code repository.

Can you explain what in that Trello card is meant to cover this specific case.

**Comment by jwforres** [comment]
in general that card is the catch-all for supporting the build options we are missing, including picking your own builders (that used to be explicitly mentioned in the card).  I added it to the line item where i had linked this issue.
